### DISCUSSION TITLE
Critères de sélection et autres résolveurs de DNS

### DISCUSSION THREAD

**Discussion by bortzmeyer** [changetondns/changetondns.fr#15] [General]
Sur quels critères sont choisis les résolveurs listés à la fin de la page ? Il serait bon de les publier. (C'est lié à #6 et #7 .)

**Comment by ostix360** [comment]
C'est au utilisateur de choisir mais effectivement il manque des choses sur le site.
Comme critère il peut y avoir:
- la rapidité 
- l'accès aux site existant (le dns ne doit pas mentir sur l'existence ou non d'un site)
- la suppression des données utilisateur lors d'une requête.
Qu'en pensez vous et y en a t il d'autres?

**Comment by bortzmeyer** [comment]
> C'est au utilisateur de choisir mais effectivement il manque des choses sur le site. Comme critère il peut y avoir:
> 
>     * la rapidité
> 
>     * l'accès aux site existant (le dns ne doit pas mentir sur l'existence ou non d'un site)
> 
>     * la suppression des données utilisateur lors d'une requête.
>       Qu'en pensez vous et y en a t il d'autres?

- nationalité (les promesses ne valent rien si la loi impose des les violer, comme avec le Cloud Act aux USA)
- caractère lucratif ou associatif
- services offerts (DoT, DoH, etc)

Et, non, ce n'est pas l'utilisateur qui choisit, à l'heure actuelle, c'est l'administrateur du site qui, parmi les dizaines de résolveurs publics existants, a choisi une poignée d'entre eux.

**Comment by ostix360** [comment]
L'objectif avec ce site est, que l'utilisateur choisisse son résolveur de DNS par rapport à ses critères et non pas qu'il subisse celui qu'il lui a été attribué par défaut.
Les autres critères semble être tout autant pertinent.
Merci bien

**Comment by bortzmeyer** [comment]
OK, il y a manifestement un problème de compréhension, @ostix360 est parti sur tout à fait autre chose, qui n'a pas de rapport avec mon ticket. Je reformule donc : le site ne publie actuellement pas tous les résolveurs DNS connus mais une sélection. Sur quels critères a été faite cette sélection ?

**Comment by Jourdelune** [comment]
On a juste pris des DNS populaire, mais cette sélection peut être peaufiné ou autre, seulement on a considéré qu'afficher trop de DNS pouvait nuir à l'utilisateur (quel DNS il doit choisir parmit tant, 100 par exemple ?). Seulement si vous considérez que les DNS affichés sont mauvais ou autre et que vous souhaitez en ajouter, vous pouvez faire des propositions. Cette liste peut être amené à totalement évoluer. Nous ne sommes pas attaché à des DNS spécifiques (on a mis ceux qu'on utilisais et qui fonctionnait correctement mais il est certains qu'il y en a d'autre qui peuvent même être plus pertinent et qui ont été oublié).

**Comment by matrixbx** [comment]
Je rejoins Stéphane, proposer sans avertir des problèmes de confidentialité (à minima) les DNS de Google, Cloudflare et consort, c'est très "problématique", il me semble préférable de mettre en avant ceux proposés par des structures Fr non lucratives en l'expliquant.

Et tant qu'à y être, expliquer qu'il est également préférable d'en choisir au moins deux pas chez le même fournisseur pour être un peu plus résilient.

Voir également https://diyisp.org/dokuwiki/doku.php?id=technical:dnsresolver

**Comment by Jourdelune** [comment]
En effet, il est peut-être intéressant de supprimer le DNS de google mais garder cloudflare sans l'afficher en premier. Concernant la description des DNS dans le tableau vous pouvez faire des suggestions pour les changer, nous avons mis celle-ci car elles nous semblaient refléter la réalité du résolveur DNS mais nous sommes ouvert à tous changement.

**Comment by ostix360** [comment]
Je me suis permis de modifier le titre et de transformer l'issue en discussion pour que ce soit plus large comme discussion.
C'est pourquoi je vous invite à discuter des différents ajouts et modification concernant le tableau de DNS

**Comment by matrixbx** [comment]
Concernant l'argument de la rapidité, les résolveurs du FAIs sont très généralement les plus proches d'un point de vu strictement réseau, ils bénéficient donc généralement de la plus faible latence d'accès, reste les temps de traitement des requêtes qu'il faut évaluer/mesurer en essayant.
À préciser que plus un résolveurs est utilisé par de nombreuses personnes et plus la probabilité que la réponse soit déjà en cache est élevée et donc le temps de réponse réduit (pas besoin d'aller solliciter les serveurs faisant autorité).

Bien préciser que le choix d'un résolveur peut éventuellement avoir une incidence sur la navigation, le temps de chargement des pages dépendant d'abord du temps de résolution des noms de domaines utilisés, mais pas sur le débit de téléchargement.

Impérativement expliquer que les résolveurs ont au moins potentiellement une idée très très précise de tout les sites visités en soulignant qu'on leur confie donc une part importante d'informations très personnelles et donc privées que certains n'hésiteront pas à monétiser de toutes les manières envisageables.
